ui/desktop/app/controllers/application.js

```diff
@@ -75,4 +102,34 @@ export default class ApplicationController extends Controller {
         rootEl.classList.remove('rose-theme-light');
     }
   }
+
+  @action
+  signoutAttempt() {
+    this.showSignoutModal = true;
+    // Prevents a user from quitting app while signout modal is present
+    this.ipc.invoke('setSignoutInProgress', true);
+  }
+
+  @action
+  cancelSignout() {
```

Comment:
You could consolidate these cancel functions to just set both variables to false to use one modal